### PR TITLE
Winflex john hancock protection iul 15 config

### PR DESCRIPTION
**Pull request by 7seven7lst**
I've generated a config file with directives for John Hancock protection IUL 15 plan.
those will be the user data
```
{
  "user": {
    "full_name": "Test User",
    "gender": "M",
    "age": "30",
    "state": "CA",
    "class": "SP"
  },
  "agent": {
    "full_name": "Test Agent Name",
    "company": "Test Agent Company",
    "address1": "123 Foo Bar",
    "state": "CA",
    "zipcode": "92111"
  }
}
```

@escoffon Please let me know if this looks correct to you.

### CODE REVIEW

File: winflex/config/JohnHancock-protection-iul-15.cfg [hunk 1]
```diff
@@ -0,0 +1,1410 @@
...
+      "value": {
+        "Insured": {
+          "Name": {
+            "keyword": "Insured.Name",
+            "type": "normal",
+            "value": "Stan List",
+            "description": "",
+            "use": ":user.full_name",
+            "useType": "String"
+          },
```
Comment (7seven7lst) [inline winflex/config/JohnHancock-protection-iul-15.cfg]: I was a little unsure if we should erase the info in the `"value"` and `"descriptoin"` field above?

Comment (escoffon) [reply winflex/config/JohnHancock-protection-iul-15.cfg]: No leave them in place. That way we have an explicit statement as to what was in the original. But you could set 'value' to an empty value where you use directives (and for the ClientInfo section it is 'answer')

Comment (7seven7lst) [reply winflex/config/JohnHancock-protection-iul-15.cfg]: ah. Gotcha. sounds good. 👍 
basically all the `"value"` and `"answer"` will be overriden by the "use" value, and `"description"` is not really used (like a label or something).
I will leave the "value" as it is. thx

**Comment by 7seven7lst** [comment]
@escoffon I'm getting used to the process now and it's pretty cool. Pretty easy to use and great job there 🥇 

What I learned from the first couple tries:
* I figured that I can miss some fields when navigating through the config file and replace with directives. For that, I think I will try use a easily searchable sequence for the input (`"i.e. name: John Doe SEARCHABLE"`)
* I will keep a screenshot of the field that I need to enter, and the warnings(errors). so I can try to resubmit the illustration and/or remember which fields are required. 

Other than that, I think it's pretty straightforward now I get used to it. thx for writing the documentation on that and explaining in details.

File: winflex/config/JohnHancock-protection-iul-15.cfg (hunk 1 above)
Comment (escoffon) [reply winflex/config/JohnHancock-protection-iul-15.cfg]: all the values and answer where 'use' is not empty. Otherwise, we leave it as is.

**Comment by escoffon** [comment]
One thing I did that may or may not help is I grouped all elements in a hierarchy. The request data are sent down as a flat array, but in reality the data do have some structure. The code that generates the config file extracts that structure and instead generates a hierarchy where data with a common "root" are grouped together. I thought it might make it easy to find related data components, since in this layout they are all close by, whereas in an array there is no guarantee of that. And it might help developers to get an idea of the data organization. But, it could turn out to be confusing instead. I'm not 100% sure that this was the way to go, so let me know if you find the hierarchical organization cumbersome.

**Comment by 7seven7lst** [comment]
@escoffon 
I like the hierarchical organization data structure. It's definitely easier than the array. It's a lot easier to narrow down where the field locates, and overall the data organization is also more clear

**Pull request merged**